src/Compiler/TypedTree/TcGlobals.fs

```diff
@@ -1278,6 +1278,8 @@ type TcGlobals(
 
   member val ArrayCollector_tcr = mk_MFCompilerServices_tcref fslibCcu "ArrayCollector`1"
 
+  member val SupportsWhenTEnum_tcr = mk_MFCompilerServices_tcref fslibCcu "SupportsWhenTEnum"
```

Comment:
What happens when an older Fsharp.Core is explicitely added instead of the implicit one?
Does this value support optionality?

Reply:
Yes, this does still work if the FSharp.Core version referenced does not contain this type (just as with `ListCollector` and `ArrayCollector` above).